addons/product/static/src/js/pricelist_report/product_pricelist_report.js

```diff
@@ -0,0 +1,213 @@
+            "message": message
+        },
+    }
+}
+
+export class ProductPricelistReport extends Component {
+    setup() {
+        this.action = useService("action");
+        this.orm = useService("orm");
+
+        this.MAX_QTY = 5;
+        const pastState = this.props.state || {};
+
+        this.activeIds = this.props.action.context.active_ids;
+        this.activeModel = this.props.action.context.active_model;
+
+        this.state = useState({
+            displayPricelistTitle: pastState.displayPricelistTitle || false,
+            html: "",
+            pricelists: [],
+            _quantities: pastState.quantities || [1, 5, 10],
+            selectedPricelist: {},
+        });
+
+        onWillStart(async () => {
+            this.state.pricelists = await this.getPricelists()
+            this.state.selectedPricelist = pastState.selectedPricelist || this.pricelists[0];
+
+            this.renderHtml();
+        });
+
+        onRendered(() => {
+            this.env.config.setDisplayName(this.env._t("Pricelist Report"));
+        });
+
+        /*
+        When following the link of a product and coming back we need to keep the 
+        precedent state:
+            - if the pricelist was being showed
+            - wich pricelist is selected at the moment
+            - which quantities
+        */
+        useSetupAction({
+            getLocalState: () => {
+                return {
+                    displayPricelistTitle: this.displayPricelistTitle,
+                    quantities: this.quantities,
+                    selectedPricelist: this.selectedPricelist,
+              };
+            },
+        });
+    }
+
+    // getters and setters
+
+    get displayPricelistTitle() {
+        return this.state.displayPricelistTitle;
+    }
+
+    get html() {
+        return this.state.html;
+    }
+
+    get pricelists() {
+        return this.state.pricelists;
+    }
+
+    get quantities() {
+        return this.state._quantities;
+    }
+
+    set quantities(value) {
+        this.state._quantities = value;
+    }
+
+    get reportParams() {
+        return {
+            active_model: this.activeModel || 'product.template',
+            active_ids: this.activeIds || [],
+            display_pricelist_title: this.displayPricelistTitle || '',
+            pricelist_id: this.selectedPricelist.id || '',
+            quantities: this.quantities || [1],
+        };
+    }
+
+    get selectedPricelist() {
+        return this.state.selectedPricelist;
+    }
+
+    // orm calls
+
+    getPricelists() {
+        return this.orm.searchRead("product.pricelist", [], ["id", "name"]);
+    }
+
+    async renderHtml() {
+        let html = await this.orm.call(
+            "report.product.report_pricelist", "get_html", [], {data: this.reportParams}
+        );
+        this.state.html = markup(html);
+    }
+
+    // events
+
+    async onClickAddQty(ev) {
+        ev.preventDefault(); // avoid automatic reloading of the page
```

Comment:
t-on-click.prevent=
in the template

Reply:
I will keep it in mind for the future, thanks :)